net

```diff
@@ -1113,21 +1113,27 @@ def do_connect(conn, psk=None):
     set_mac(iface, mac)
     set_dns(dns)
 
-    if not wpa:
-        if is_wifi_iface(iface):
-            print 'Connecting to %s' % ssid
-            cmd = 'iwconfig %s key open ap %s' % (iface, ap_addr or 'any')
-            if ssid:
-                escaped_ssid = ''.join("\\x%02x" % ord(c) for c in ssid)
-                cmd += ' essid -- "%s"' % escaped_ssid
-            else:
-                cmd += ' essid any'
-            run(cmd)
+    # If wired
+    if not is_wifi_iface(iface):
+        if is_wifi:
+            die('WiFi options set on wired interface!')
         if not addr:
             print 'Sending DHCP request'
         for cmd in network:
             run_or_die(cmd)
+    # Else WiFi
     else:
+        # If Open WiFi with fallthrough--always use wpa_supplicant
```

Comment:
Why do you want to use wpa_supplicant for open networks?

Reply:
Got rid of `iwconfig`, and easy fix versus changed flow to connect with `iw`--which might be a fix for the future :)
Also; contemplating removing dependency on `iw` and only use `wpasupplicant`.